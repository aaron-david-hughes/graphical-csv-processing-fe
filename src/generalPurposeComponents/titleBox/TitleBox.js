import React from 'react';

class TitleBox extends React.Component {

    render() {
        return <div
            id={this.props.id}
            style={this.props.style}
        >
            <div
                id={this.props.id + '-header'}
                onClick={this.props.isCollapsible}
                style={{
                    display: 'inline-flex',
                    width: '100%',
                    justifyContent: 'space-between',
                    border: '2px solid darkslategray',
                    borderTopLeftRadius: '10px',
                    borderTopRightRadius: '10px',
                    borderBottom: this.props.open ? '0px' : '2px solid darkslategray',
                    borderBottomLeftRadius: this.props.open ? '0px' : '10px',
                    borderBottomRightRadius: this.props.open ? '0px' : '10px',
                    backgroundColor: 'white'
                }}
            >
                <h3 style={{marginLeft: '1rem'}}>{this.props.title}</h3>
                {
                    this.props.icon
                    ? <button
                        style={{
                            border: 'none'
                        }}
                        onClick={this.props.iconOnClick}
                        title={this.props.iconTitle}
                    >
                        {this.props.icon}
                    </button>
                    : null
                }
            </div>
            <div
                id={this.props.id + '-body'}
                style={{
                    display: this.props.open ? 'block' : 'none',
                    width: '100%',
                    border: '2px solid darkslategray',
                    borderBottomLeftRadius: '10px',
                    borderBottomRightRadius: '10px',
                    backgroundColor: 'white',
                }}
            >
                <div style={{margin: '1rem'}}>
                    {this.props.children}
                </div>
            </div>
        </div>
    }
}

export default TitleBox;