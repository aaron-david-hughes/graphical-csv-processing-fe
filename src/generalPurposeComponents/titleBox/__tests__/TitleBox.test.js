import React from 'react';
import {shallow} from 'enzyme';
import TitleBox from "../TitleBox";

describe('TitleBox component tests', () => {
    let props = {
        id: 'testTitleBoxId',
        style: 'testStyle',
        isCollapsible: true,
        open: true,
        title: 'testTitleBoxTitle',
        icon: 'testIcon',
        iconOnClick: jest.fn(),
        iconTitle: 'testIconTitle',
        children: 'titleBoxContent'
    };

    let render;

    beforeEach(() => {
        render = shallow(<TitleBox {...props} />);
    });

    describe('render tests', () => {
        it('should match open snapshot when open is true', () => {
            expect(render).toMatchSnapshot();
        });

        it('should match closed snapshot when open is false', () => {
            render = shallow(<TitleBox {...props} open={false} />);
            expect(render).toMatchSnapshot();
        });

        it('should call iconOnClick function when button is clicked', () => {
            render.find('button').simulate('click');

            expect(props.iconOnClick).toHaveBeenCalled();
        });
    });
});