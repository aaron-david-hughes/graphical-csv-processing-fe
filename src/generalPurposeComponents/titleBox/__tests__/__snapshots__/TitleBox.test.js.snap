// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TitleBox component tests render tests should match closed snapshot when open is false 1`] = `
<div
  id="testTitleBoxId"
  style="testStyle"
>
  <div
    id="testTitleBoxId-header"
    onClick={true}
    style={
      Object {
        "backgroundColor": "white",
        "border": "2px solid darkslategray",
        "borderBottom": "2px solid darkslategray",
        "borderBottomLeftRadius": "10px",
        "borderBottomRightRadius": "10px",
        "borderTopLeftRadius": "10px",
        "borderTopRightRadius": "10px",
        "display": "inline-flex",
        "justifyContent": "space-between",
        "width": "100%",
      }
    }
  >
    <h3
      style={
        Object {
          "marginLeft": "1rem",
        }
      }
    >
      testTitleBoxTitle
    </h3>
    <button
      onClick={[MockFunction]}
      style={
        Object {
          "border": "none",
        }
      }
      title="testIconTitle"
    >
      testIcon
    </button>
  </div>
  <div
    id="testTitleBoxId-body"
    style={
      Object {
        "backgroundColor": "white",
        "border": "2px solid darkslategray",
        "borderBottomLeftRadius": "10px",
        "borderBottomRightRadius": "10px",
        "display": "none",
        "width": "100%",
      }
    }
  >
    <div
      style={
        Object {
          "margin": "1rem",
        }
      }
    >
      titleBoxContent
    </div>
  </div>
</div>
`;

exports[`TitleBox component tests render tests should match open snapshot when open is true 1`] = `
<div
  id="testTitleBoxId"
  style="testStyle"
>
  <div
    id="testTitleBoxId-header"
    onClick={true}
    style={
      Object {
        "backgroundColor": "white",
        "border": "2px solid darkslategray",
        "borderBottom": "0px",
        "borderBottomLeftRadius": "0px",
        "borderBottomRightRadius": "0px",
        "borderTopLeftRadius": "10px",
        "borderTopRightRadius": "10px",
        "display": "inline-flex",
        "justifyContent": "space-between",
        "width": "100%",
      }
    }
  >
    <h3
      style={
        Object {
          "marginLeft": "1rem",
        }
      }
    >
      testTitleBoxTitle
    </h3>
    <button
      onClick={[MockFunction]}
      style={
        Object {
          "border": "none",
        }
      }
      title="testIconTitle"
    >
      testIcon
    </button>
  </div>
  <div
    id="testTitleBoxId-body"
    style={
      Object {
        "backgroundColor": "white",
        "border": "2px solid darkslategray",
        "borderBottomLeftRadius": "10px",
        "borderBottomRightRadius": "10px",
        "display": "block",
        "width": "100%",
      }
    }
  >
    <div
      style={
        Object {
          "margin": "1rem",
        }
      }
    >
      titleBoxContent
    </div>
  </div>
</div>
`;
